pkg/reconciler/openshift/tektonaddon/tektonaddon.go

```diff
@@ -369,26 +354,34 @@ func (r *Reconciler) ensureClusterTasks(ctx context.Context, ta *v1alpha1.Tekton
 // and if installer set which already exist is of older version then it deletes and return false to create a new
 // installer set
 func checkIfInstallerSetExist(ctx context.Context, oc clientset.Interface, relVersion string,
-	ta *v1alpha1.TektonAddon, component string) (bool, error) {
+	component string) (bool, error) {
```

Comment:
the instead of sending the component, version etc separately the last paramerte should be `labelSelector` so that whoever using this function could format `1` or `n` labels and pass it to this function. 

Otherwise, we will have to change the the implementation of this function for each use case for other components in the future.